libr/core/cmd_print.c

```diff
@@ -2431,10 +2434,15 @@ r_cons_pop();
 			if (show_comments) {
 				char *comment = r_core_anal_get_comments (core, addr);
 				if (comment) {
-					if (show_offset) {
-						r_cons_printf ("%s0x%08"PFMT64x" ", use_color? pal->offset: "", addr);
+					if (strcmp(comment, switchcmp)) {
+						if (show_offset) {
+							r_cons_printf ("%s0x%08"PFMT64x" ", use_color? pal->offset: "", addr);
+						}
+						r_cons_printf ("%s%s\n", use_color? pal->comment: "", comment);
+					}
+					if (!strncmp (comment, "switch table", 12)) {
+						switchcmp = strdup (comment);
 					}
-					r_cons_printf ("%s%s\n", use_color? pal->comment: "", comment);
 					R_FREE (comment);
 				}
 			}
```

Comment:
free it before leaving

Reply:
switchcmp should be freed at the end of the for loop (or function), since I want it to retain its value each time a new comment is processed, right?